src/Catalog/Helpers/GalleryDatabaseHelper.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using NuGet.Services.Sql;
+
+namespace NuGet.Services.Metadata.Catalog.Helpers
+{
+    /// <summary>
+    /// Helper methods for accessing and reading from the gallery database.
+    /// </summary>
+    public class GalleryDatabaseHelper : IGalleryDatabaseHelper
+    {
+        private readonly ISqlConnectionFactory _connectionFactory;
+        private readonly IDb2CatalogQueries _db2catalogQueries;
+
+        public GalleryDatabaseHelper(
+            ISqlConnectionFactory connectionFactory,
+            IDb2CatalogQueries db2catalogQueries)
+        {
+            _connectionFactory = connectionFactory ?? throw new ArgumentNullException(nameof(connectionFactory));
+            _db2catalogQueries = db2catalogQueries ?? throw new ArgumentNullException(nameof(db2catalogQueries));
+        }
+
+        public Task<SortedList<DateTime, IList<FeedPackageDetails>>> GetPackagesCreatedSince(DateTime since, int top)
+        {
+            return GetPackagesInOrder(
+                since,
+                top,
+                package => package.CreatedDate,
+                "Created");
+        }
+
+        public Task<SortedList<DateTime, IList<FeedPackageDetails>>> GetPackagesEditedSince(DateTime since, int top)
+        {
+            return GetPackagesInOrder(
+                since,
+                top,
+                package => package.LastEditedDate,
+                "LastEdited");
+        }
+
+        /// <summary>
+        /// Returns a <see cref="SortedList{DateTime, IList{FeedPackageDetails}}"/> from the gallery database.
+        /// </summary>
+        /// <param name="keyDateFunc">The <see cref="DateTime"/> field to sort the <see cref="FeedPackageDetails"/> on.</param>
+        private async Task<SortedList<DateTime, IList<FeedPackageDetails>>> GetPackagesInOrder(
+            DateTime since,
+            int top,
+            Func<FeedPackageDetails, DateTime> keyDateFunc,
+            string columnName)
+        {
+            var allPackages = await GetPackages(since, top, columnName);
+
+            return OrderPackagesByKeyDate(allPackages, keyDateFunc);
+        }
+
+
+        /// <summary>
+        /// Returns a <see cref="SortedList{DateTime, IList{FeedPackageDetails}}"/> of packages.
+        /// </summary>
+        /// <param name="keyDateFunc">The <see cref="DateTime"/> field to sort the <see cref="FeedPackageDetails"/> on.</param>
+        public static SortedList<DateTime, IList<FeedPackageDetails>> OrderPackagesByKeyDate(
```

Comment:
UnitTest?

Reply:
This is existing code which lacked test coverage. Added a test asserting behavior in https://github.com/NuGet/NuGet.Services.Metadata/pull/524/commits/67cd6d2138b28d6fe984e4c27be3c1def5df9bf1.